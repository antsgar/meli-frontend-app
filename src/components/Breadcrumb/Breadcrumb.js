import styles from "./Breadcrumb.module.scss";

const Breadcrumb = ({ breadcrumbItems }) => {
    const items = breadcrumbItems.map((item) => (
        <span key={item} className={styles.item}>
            {item}
        </span>
    ));

    return <div className={styles.container}>{items}</div>;
};

export default Breadcrumb;
