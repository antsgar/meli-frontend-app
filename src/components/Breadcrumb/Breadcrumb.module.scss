@import "../../styles/vars";

.container {
    min-height: 16px;
    margin-top: $spacing-medium;
    margin-bottom: $spacing-medium;
}

.item {
    font-size: 16px;
    color: $text-color-light;

    &:not(:last-child)::after {
        content: ">";
        margin-left: $spacing-small;
        margin-right: $spacing-small;
    }

    &:last-child {
        font-weight: bold;
    }

    &:hover {
        opacity: 0.8;
        cursor: pointer;
    }
}
