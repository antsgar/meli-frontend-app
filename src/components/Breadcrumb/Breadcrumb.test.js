import React from "react";
import { mount } from "enzyme";
import { categories } from '../../fixtures';
import Breadcrumb from './Breadcrumb';

let wrapper;

beforeEach(() => {
    wrapper = mount(<Breadcrumb breadcrumbItems={categories} />);
});

describe("Breadcrumb", () => {
    it("should render component", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("should render items", () => {
        const renderedItems = wrapper.find('.item');
        expect(renderedItems).toHaveLength(categories.length);
        categories.map((category, index) => {
            expect(renderedItems.at(index).text()).toEqual(category);
        });
    });
});
