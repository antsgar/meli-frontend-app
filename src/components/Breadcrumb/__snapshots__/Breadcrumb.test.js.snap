// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb should render component 1`] = `
<Breadcrumb
  breadcrumbItems={
    Array [
      "Celulares y Teléfonos",
      "Celulares y Smartphones",
    ]
  }
>
  <div
    className="container"
  >
    <span
      className="item"
      key="Celulares y Teléfonos"
    >
      Celulares y Teléfonos
    </span>
    <span
      className="item"
      key="Celulares y Smartphones"
    >
      Celulares y Smartphones
    </span>
  </div>
</Breadcrumb>
`;
